client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/SubmissionEditForm.jsx

```diff
@@ -475,30 +481,48 @@ const SubmissionEditForm = (props) => {
     />
   );
 
-  const renderTabbedQuestions = () => (
-    <Tabs
-      onChange={(event, value) => {
-        setStepIndex(value);
-      }}
-      style={{ backgroundColor: grey[100], color: blue[500] }}
-      TabIndicatorProps={{ color: 'primary', style: { height: 5 } }}
-      value={stepIndex}
-      variant="fullWidth"
+  const renderSteppedQuestions = () => (
+    <Stepper
+      activeStep={stepIndex}
+      connector={<div />}
+      nonLinear
+      style={{ justifyContent: 'center', flexWrap: 'wrap', padding: 10 }}
     >
-      {questionIds.map((id, index) => (
-        <Tab
-          key={id}
-          label={intl.formatMessage(translations.questionNumber, {
-            number: index + 1,
-          })}
-          style={{ minWidth: 10 }}
-          value={index}
-        />
-      ))}
-    </Tabs>
+      {questionIds.map((id, index) => {
+        let stepButtonColor = '';
+        const isCurrentQuestion = index === stepIndex;
+        stepButtonColor = isCurrentQuestion ? blue[800] : blue[400];
+        return (
+          <Step key={id} sx={{ width: 55, height: 50 }}>
+            <StepButton
+              icon={
+                <SvgIcon htmlColor={stepButtonColor} sx={{ fontSize: '2.5em' }}>
+                  <circle cx="12" cy="12" r="12" />
+                  <text
+                    x="12"
+                    y="16"
+                    textAnchor="middle"
+                    fontSize="10"
+                    fill="#fff"
+                  >
+                    {index + 1}
+                  </text>
+                </SvgIcon>
+              }
+              onClick={() => {
+                if (published || graderView) {
```

Comment:
Why do we need this check?

Reply:
Oh that shouldnt be there it blocks students from changing questions